tests/test_banner.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Minitest for Banner class
+#
+# Copyright (c) 2014-2018 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/banner'
+
+# TestBanner - Minitest for Banner node utility.
+class TestBanner < CiscoTestCase
+  def setup
+    # setup runs at the beginning of each test
+    super
+    no_motd
+  end
+
+  def teardown
+    # teardown runs at the end of each test
+    no_motd
+    super
+  end
+
+  def no_motd
+    # Turn the feature off for a clean test.
+    config('no banner motd')
+  end
+
+  # TESTS
+
+  def test_motd
```

Comment:
This test fails on my devices.

```shell
Node under test:
  - name    - n9k
  - type    - N9K-C9504
  - image   - bootflash:///nxos.9.2.1.bin
  - version - 9.2(1)

TestBanner#test_motd = 3.62 s = F

Finished in 3.625701s, 0.2758 runs/s, 1.1032 assertions/s.

  1) Failure:
TestBanner#test_motd [tests/test_banner.rb:49]:
Expected "User Access Verification\n" to be nil.

1 runs, 4 assertions, 1 failures, 0 errors, 0 skips
```

The behavior on my devcies is as follows.

```
n9k# conf t
Enter configuration commands, one per line. End with CNTL/Z.
n9k(config)# no banner motd
n9k(config)# 
n9k(config)# end
n9k# 
n9k# show banner motd 
User Access Verification
n9k# 
```

Reply:
Or.. why not check that it is set to the default value?

Reply:
Good catch - missed updating this when I changed the behavior :)